hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/IOUtils.java

```diff
@@ -119,6 +119,9 @@ public static Iterator<List<WriteStatus>> runMerge(HoodieMergeHandle<?, ?, ?, ?>
           "Error in finding the old file path at commit " + instantTime + " for fileId: " + fileId);
     } else {
       mergeHandle.doMerge();
+      if (mergeHandle instanceof FileGroupReaderBasedMergeHandle) {
+        mergeHandle.close();
```

Comment:
Open question: Is there any reason to avoid calling close on the other merge handles?

Reply:
It seems we shoud just return `mergeHandle.close` in line 132 instead of closing here for all the merge handles.